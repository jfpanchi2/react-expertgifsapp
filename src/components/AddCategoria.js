import React, { useState } from 'react'
import PropTypes from 'prop-types';
export const AddCategoria = ({setCategorias}) => {
    const [inputValue, setinputValue] = useState('');
    const hanldeInputChange = (e) => {
        //console.log(e.target.value);
        setinputValue(e.target.value);
    } 
    
    const handleSubmit =(e) =>{
        e.preventDefault();
        if(inputValue.trim().length>2){
            setCategorias(categorias=>[inputValue,...categorias]);
        }
        setinputValue("");
        //console.log('Submit exitoso');
       
    }
    return (
        <form onSubmit={handleSubmit}>
            <h2>Buscar gifs</h2>
            <input 
            type="text" 
            value={inputValue}
            onChange={hanldeInputChange}></input>
        </form>
        
    )
}

AddCategoria.propTypes ={
    setCategorias: PropTypes.func.isRequired
}