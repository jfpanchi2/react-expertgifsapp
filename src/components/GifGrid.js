import React  from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
export const GifGrid = ({categoria}) => {
    /* const [imagen, setimagen] = useState([]); */
    /* const [count, setcount] = useState(0); */
    
    const {data:imagenes,loading} = useFetchGifs(categoria)
    
    /* useEffect(()=>{
        getGifs(categoria)
        .then(imagenes =>setimagen(imagenes));
    },[categoria]) */
    
    //getGifs();
    return (
        <div>
            <h3 className="animate__animated animate__fadeIn">{categoria}</h3>
            {loading && <h4 className="animate__animated animate__flash">Cargando...</h4>}
            <div className="card-grid">
            
                {
                        imagenes.map(img =>{
                           return <GifGridItem key={img.id} {...img} ></GifGridItem>
                        })
                }        
            
           {/*  <h3>{count}</h3>
            <button onClick={()=> setcount(count +1)}></button> */}
            </div>
        </div>
        
    )
}
