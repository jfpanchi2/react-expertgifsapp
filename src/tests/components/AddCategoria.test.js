import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { AddCategoria } from '../../components/AddCategoria';

describe('Pruebas en AddCategoria', () => {
    const setCategories = jest.fn();
    let wrapper;
    
    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategoria setCategorias={ setCategories}></AddCategoria>);
    });

    test('Debe de mostarse correctamente', () => {
       
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo'
        input.simulate('change',{target: {value}});
        expect(wrapper.find('p').text().trim()).toBe(value);
    })
    
    test('No debe de psotear la informaciÃ³n con Submit', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('Debe de llamar el setCategories y limpiar caja', () => {
        const value = 'Hola mundo';
        // 1. Simular el inputChange
        wrapper.find('input').simulate('change',{target:{value}});
        // 2. Simular el submit 
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        // 3. set caterias se debe llamar
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        // 4. el valor de input debe ser ''
        expect(wrapper.find('input').prop('value')).toBe('');
    })
    
    
})
