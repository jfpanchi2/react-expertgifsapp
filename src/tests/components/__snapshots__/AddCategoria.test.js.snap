// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategoria Debe de mostarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <h2>
    Buscar gifs
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
